src/slacko.ml

```diff
@@ -1075,8 +1135,12 @@ let chat_delete token ts chat =
     | #message_error as res -> res
     | _ -> `Unknown_error
 
+let jsonify_attachments attachments =
+  `List (List.map (fun a -> attachment_obj_to_yojson a) attachments)
+  |> Yojson.Safe.to_string
+
 let chat_post_message token chat
-  ?username ?parse ?icon_url ?icon_emoji text =
+  ?username ?parse ?icon_url ?icon_emoji ?attachments text =
```

Comment:
It would probably be more consistent to default `attachments` to `[]` and then `definitely_add` them, so there is no ambiguity and different behaviour between `Some []` and `None`.

Unless the Slack API does something weird when attachments are added but said list is empty. In such case it would probably still make sense to dispatch behaviour inside of this function and do not expose that in the signature.